### PR TITLE
Feature/MDS-2893 Create Permit Conditions

### PR DESCRIPTION
**Pull request by nayr974**
User Story:
As a permit writer, I would like to create conditions under a selected category, I would also like to create sub-sections and list items under those sub-sections so that I can later manage and organize permits in a  way that is meaningful to proponents and consistent across the province. 

AC
Ability to create a Subsection - with an input field for title value 
Ability to add conditions under the sub-section
Ability to add a list item to a selected condition

See Frame 2-12 : https://www.figma.com/file/JwmaVHGXAXQc8YJqyNqXIDEn/MDS---CORE-%26-Minespace?node-id=3091%3A1440

### CODE REVIEW

File: services/core-web/src/components/Forms/permits/conditions/Condition.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import Section from "@/components/Forms/permits/conditions/Section";
+import ListItem from "@/components/Forms/permits/conditions/ListItem";
+
+const propTypes = {
+    condition: PropTypes.objectOf(PropTypes.any)
+};
+
+const defaultProps = {
+};
+
```
Comment (mmumby) [inline services/core-web/src/components/Forms/permits/conditions/Condition.js:14]: children should be in defaultProps if not required

File: services/core-web/src/components/Forms/permits/conditions/AddCondition.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    editingConditionFlag: PropTypes.bool.isRequired,
+    setEditingConditionFlag: PropTypes.func.isRequired,
+    createPermitCondition: PropTypes.func.isRequired,
+    fetchPermitConditions: PropTypes.func.isRequired,
+    draftPermitAmendment: CustomPropTypes.permit.isRequired,
+};
+
+const defaultProps = {
+};
+
```
Comment (mmumby) [inline services/core-web/src/components/Forms/permits/conditions/AddCondition.js:25]: defaultProps can be removed if all are required

File: services/core-web/src/components/Forms/permits/conditions/Conditions.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    permitConditionTypeOptions: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
+    fetchPermitConditions: PropTypes.func.isRequired,
+    noticeOfWork: CustomPropTypes.importedNOWApplication.isRequired,
+    draftPermitAmendment: CustomPropTypes.permit.isRequired,
+    setEditingConditionFlag: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+};
+
```
Comment (mmumby) [inline services/core-web/src/components/Forms/permits/conditions/Conditions.js:33]: same here. Are you not getting linting errors for these when committing?

File: services/core-web/src/components/Forms/permits/conditions/ListItem.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    condition: PropTypes.objectOf(PropTypes.any),
+    new: PropTypes.bool,
+    handleSubmit: PropTypes.func,
+    handleCancel: PropTypes.func,
+    initialValues: PropTypes.objectOf(PropTypes.any),
+};
+
+const defaultProps = {
+    condition: {
+        step: '',
```
Comment (mmumby) [inline services/core-web/src/components/Forms/permits/conditions/ListItem.js:19]: What is condition.step referring to?

File: services/core-api/app/api/exports/static_content/resources/core_static_content_resource.py [hunk 5]
```diff
@@ -74,7 +76,8 @@ class StaticContentResource(Resource):
     )
     @requires_role_view_all
     def get(self):
-        content_json = cache.get(STATIC_CONTENT_KEY)
+        # content_json = cache.get(STATIC_CONTENT_KEY)
+        content_json = None
         if not content_json:
```
Comment (pdittaro) [inline services/core-api/app/api/exports/static_content/resources/core_static_content_resource.py:81]: Is this a placeholder?

File: services/core-api/app/api/mines/namespace.py [hunk 6]
```diff
@@ -34,6 +34,9 @@
 from app.api.mines.variances.resources.variance_uploaded_documents import MineVarianceUploadedDocumentsResource
 from app.api.parties.party_appt.resources.mine_party_appt_document_upload_resource import MinePartyApptDocumentUploadResource
 from app.api.mines.comments.resources.mine_comment import MineCommentListResource, MineCommentResource
+from app.api.mines.permits.permit_conditions.resources.premit_conditions_resource import PermitConditionsListResource, PermitConditionsResource
```
Comment (pdittaro) [inline services/core-api/app/api/mines/namespace.py:37]: Typo `premit`

File: services/core-api/app/api/parties/party_appt/resources/mine_party_appt_resource.py [hunk 7]
```diff
@@ -57,6 +57,7 @@ def get(self, mine_party_appt_guid=None):
                 raise NotFound('Mine Party Appointment not found')
             result = mpa.json(relationships=relationships)
         else:
+            current_app.logger.info(request.args)
```
[the remark this answers is not in the document]
Comment (nayr974) [reply services/core-api/app/api/parties/party_appt/resources/mine_party_appt_resource.py:60]: I doubt it, removed

File: services/core-web/common/actionCreators/permitActionCreator.js [hunk 8]
```diff
@@ -189,3 +189,64 @@ export const deletePermitAmendment = (mineGuid, permitGuid, permitAmdendmentGuid
...
+      dispatch(success(reducerTypes.GET_PERMIT_CONDITIONS));
+      dispatch(permitActions.storePermitConditions(response.data));
+    })
+    .catch(() => dispatch(error(reducerTypes.GET_PERMIT_CONDITIONS)))
+    .finally(() => dispatch(hideLoading()));
+};
+
+export const createPermitCondition = (mineGuid, permitGuid, permitAmdendmentGuid, payload) => (dispatch) => {
+  dispatch(request(reducerTypes.CREATE_PERMIT_CONDITION));
+  dispatch(showLoading("modal"));
```
[the remark this answers is not in the document]
Comment (nayr974) [reply services/core-web/common/actionCreators/permitActionCreator.js:211]: Good catch, updated

File: services/core-web/common/actions/permitActions.js [hunk 9]
```diff
@@ -9,3 +9,13 @@ export const storeDraftPermits = (payload) => ({
   type: actionTypes.STORE_DRAFT_PERMITS,
   payload,
 });
+
+export const storePermitConditions = (payload) => ({
+  type: actionTypes.STORE_PERMIT_CONDITIONS,
+  payload
+});
+
```
[the remark this answers is not in the document]
Comment (nayr974) [reply services/core-web/common/actions/permitActions.js:17]: Correct :)

File: services/core-web/common/selectors/permitSelectors.js [hunk 10]
```diff
@@ -14,3 +16,18 @@ export const getDraftPermitForNOW = createSelector(
       )
     ) || {}
 );
+
+export const getDraftPermitAmendmentForNOW = createSelector(
```
[the remark this answers is not in the document]
Comment (nayr974) [reply services/core-web/common/selectors/permitSelectors.js:20]: Yes, I just updated your code to use this as well

File: services/core-web/src/components/Forms/permits/conditions/AddCondition.js [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+const ButtonText = (condition_type_code) =>
+    ({
+        'SEC': 'Add Sub-Section',
+        'CON': 'Add Condition',
+        'LIS': 'Add List Item',
+    }[condition_type_code]);
+
+export class AddCondition extends Component {
+    state = { isEditing: false }
+    handleSubmit = (values) => this.props.createPermitCondition(null, null, this.props.draftPermitAmendment.permit_amendment_guid, values).then(() => {
```
[the remark this answers is not in the document]
Comment (nayr974) [reply services/core-web/src/components/Forms/permits/conditions/AddCondition.js:42]: They're required for the namespace, but not used and I didn't have easy access for them here. Ideally we do have them, but it makes no functional difference so long as the /path/is/correct/ID :)

File: services/core-web/src/components/Forms/permits/conditions/Condition.js (hunk 1 above)
Comment (nayr974) [reply services/core-web/src/components/Forms/permits/conditions/Condition.js:14]: Added isRequired

File: services/core-web/src/components/Forms/permits/conditions/Conditions.js (hunk 3 above)
Comment (nayr974) [reply services/core-web/src/components/Forms/permits/conditions/Conditions.js:33]: I am currently getting an exception when the linter runs. I've put looking into it off, but I'll get it linted before EOD

File: services/core-web/src/components/Forms/permits/conditions/AddCondition.js (hunk 2 above)
Comment (nayr974) [reply services/core-web/src/components/Forms/permits/conditions/AddCondition.js:25]: removed

File: services/core-web/src/components/Forms/permits/conditions/ListItem.js (hunk 4 above)
Comment (nayr974) [reply services/core-web/src/components/Forms/permits/conditions/ListItem.js:19]: step is the dynamically generated number, letter, or roman numeral

File: services/core-api/app/api/exports/static_content/resources/core_static_content_resource.py (hunk 5 above)
Comment (nayr974) [reply services/core-api/app/api/exports/static_content/resources/core_static_content_resource.py:81]: I'm going to guess debug code form Justin, I'll revert.

File: services/core-api/app/api/mines/namespace.py (hunk 6 above)
Comment (nayr974) [reply services/core-api/app/api/mines/namespace.py:37]: this was NOT premeditated

**Pull request merged**